internal/cli/cmd/update/credentials.go

```diff
@@ -0,0 +1,202 @@
+	"github.com/akuity/kargo/internal/cli/io"
+	"github.com/akuity/kargo/internal/cli/kubernetes"
+	"github.com/akuity/kargo/internal/cli/option"
+	"github.com/akuity/kargo/internal/credentials"
+	v1alpha1 "github.com/akuity/kargo/pkg/api/service/v1alpha1"
+)
+
+type updateCredentialsOptions struct {
+	genericiooptions.IOStreams
+	*genericclioptions.PrintFlags
+
+	Config        config.CLIConfig
+	ClientOptions client.Options
+
+	Project                     string
+	Name                        string
+	Git                         bool
+	Helm                        bool
+	Image                       bool
+	Type                        string
+	RepoURL                     string
+	RepoURLPattern              string
+	Username                    string
+	Password                    string
+	ChangePasswordInteractively bool
+}
+
+func newUpdateCredentialsCommand(cfg config.CLIConfig, streams genericiooptions.IOStreams) *cobra.Command {
+	cmdOpts := &updateCredentialsOptions{
+		Config:     cfg,
+		IOStreams:  streams,
+		PrintFlags: genericclioptions.NewPrintFlags("").WithTypeSetter(kubernetes.GetScheme()),
+	}
+
+	cmd := &cobra.Command{
+		Use: `credentials [--project=project] NAME \
+    [--git | --helm | --image] \
+    [--repo-url=repo-url | --repo-url-pattern=repo-url-pattern] \
+    [--username=username] \
+    [--password=password | --interactive-password]`,
+		Aliases: []string{"credential", "creds", "cred"},
+		Short:   "Update credentials for accessing a repository",
+		Args:    cobra.ExactArgs(1),
+		Example: `
+# Update the password in my-credentials
+kargo update credentials --project=my-project my-credentials --password=my-password
+
+# Update the username in my-credentials
+kargo update credentials --project=my-project my-credentials --username=my-username
+
+# Update the credential type of my-credentials
+kargo update credentials --project=my-project my-credentials --git
+
+# Update the password in my-credentials in the default project
+kargo config set-project my-project
+kargo update credentials my-credentials --password=my-password
+
+# Update the username in my-credentials in the default project
+kargo config set-project my-project
+kargo update credentials my-credentials --username=my-username
+
+# Update the credentials type of my-credentials in the default project
+kargo config set-project my-project
+kargo update credentials my-credentials --git`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cmdOpts.complete(args)
+
+			if err := cmdOpts.validate(); err != nil {
+				return err
+			}
+
+			return cmdOpts.run(cmd.Context())
+		},
+	}
+
+	// Register the option flags on the command.
+	cmdOpts.addFlags(cmd)
+
+	// Set the input/output streams for the command.
+	io.SetIOStreams(cmd, cmdOpts.IOStreams)
+
+	return cmd
+}
+
+// addFlags adds the flags for the get credentials options to the provided
+// command.
+func (o *updateCredentialsOptions) addFlags(cmd *cobra.Command) {
+	o.PrintFlags.AddFlags(cmd)
```

Comment:
```suggestion
	o.ClientOptions.AddFlags(cmd.PersistentFlags())
	o.PrintFlags.AddFlags(cmd)
```

Reply:
Seems I overlooked these when bringing the new cmds up to date with the pattern. Will for sure fix these before merging.